src/main/java/org/spoken_tutorial/health/elasticsearch/controllers/HomeController.java

```diff
@@ -194,35 +210,85 @@ public Map<String, String> addDocument(String documentId, String documentType, S
             @PathVariable String language, @PathVariable int rank, @RequestParam String documentPath,
             @RequestParam String documentUrl, @RequestParam String view_url, @RequestParam Optional<String> category,
-            @RequestParam Optional<String> topic, @RequestParam Optional<String> outlinePath) {
+            @RequestParam Optional<Integer> categoryId, @RequestParam Optional<String> topic,
+            @RequestParam Optional<Integer> topicId, @RequestParam Optional<Integer> languageId,
+            @RequestParam Optional<String> outlinePath) {
 
-        return addDocument(documentId, documentType, documentPath, documentUrl, rank, view_url, language, category,
-                topic, outlinePath, Config.ADD_DOCUMENT);
+        return addDocument(documentId, documentType, documentPath, documentUrl, rank, view_url, language, languageId,
+                category, categoryId, topic, topicId, outlinePath, Config.ADD_DOCUMENT);
     }
 
     @PostMapping("/updateDocument/{documentId}/{documentType}/{language}/{rank}")
     public Map<String, String> updateDocument(@PathVariable String documentId, @PathVariable String documentType,
             @PathVariable String language, @PathVariable int rank, @RequestParam String documentPath,
             @RequestParam String documentUrl, @RequestParam String view_url, @RequestParam Optional<String> category,
-            @RequestParam Optional<String> topic, @RequestParam Optional<String> outlinePath) {
+            @RequestParam Optional<Integer> categoryId, @RequestParam Optional<String> topic,
+            @RequestParam Optional<Integer> topicId, @RequestParam Optional<Integer> languageId,
+            @RequestParam Optional<String> outlinePath) {
 
-        return addDocument(documentId, documentType, documentPath, documentUrl, rank, view_url, language, category,
-                topic, outlinePath, Config.UPDATE_DOCUMENT);
+        return addDocument(documentId, documentType, documentPath, documentUrl, rank, view_url, language, languageId,
+                category, categoryId, topic, topicId, outlinePath, Config.UPDATE_DOCUMENT);
     }
 
     @PostMapping("/updateDocumentRank/{documentId}/{documentType}/{language}/{rank}")
     public Map<String, String> updateDocumentRank(@PathVariable String documentId, @PathVariable String documentType,
             @PathVariable String language, @PathVariable int rank) {
-        return addDocument(documentId, documentType, null, null, rank, null, language, null, null, null,
-                Config.UPDATE_DOCUMENT_RANK);
+        return addDocument(documentId, documentType, null, null, rank, null, language, null, null, null, null, null,
+                null, Config.UPDATE_DOCUMENT_RANK);
     }
 
     @GetMapping("/deleteDocument/{documentId}/{documentType}/{language}")
     public Map<String, String> deleteDocument(@PathVariable String documentId, @PathVariable String documentType,
             @PathVariable String language) {
```

Comment:
```suggestion
            @PathVariable int languageId) {
```
change to languageId